httpbp/client_middlewares_test.go

```diff
@@ -256,11 +257,22 @@ func TestClientErrorWrapper(t *testing.T) {
 	})
 }
 
+func unwrapRetryErrors(err error) []error {
+	var errs interface {
+		error
+
+		Unwrap() []error
+	}
+	if errors.As(err, &errs) {
+		return errs.Unwrap()
+	}
+	return []error{err}
+}
+
 func TestRetry(t *testing.T) {
-	t.Run("retry for timeout", func(t *testing.T) {
```

Comment:
this subtest is removed because we don't really retry for timeouts (there's no timeout budget left for any retries), the expected attempt in this subtest is also 1.